launcher/innoextract.cpp

```diff
@@ -60,3 +60,99 @@ QString Innoextract::extract(QString installer, QString outDir, std::function<vo
+		{ H3_COMPLETE, "russian",    819416,  981633128, "e84eedf62fe2e5f9171a7e1ce6e99315a09ce41f", "49cc683395c0cf80830bfa66e42bb5dfdb7aa124" }, // setup_heroes_of_might_and_magic_3_complete_4.0_(3.2)_gog_0.1_(russian)_(77075).exe
+		{ CHR,         "english", 485694752,          0, "44e4fc2c38261a1c2a57d5198f44493210e8fc1a", ""                                         }, // setup_heroes_chronicles_chapter1_2.1.0.42.exe
+		{ CHR,         "english", 493102840,          0, "b479a3272cf4b57a6b7fc499df5eafb624dcd6de", ""                                         }, // setup_heroes_chronicles_chapter2_2.1.0.43.exe
+		{ CHR,         "english", 470364128,          0, "5ad36d822e1700c9ecf93b78652900a52518146b", ""                                         }, // setup_heroes_chronicles_chapter3_2.1.0.41.exe
+		{ CHR,         "english", 469211296,          0, "5deb374a2e188ed14e8f74ad1284c45e46adf760", ""                                         }, // setup_heroes_chronicles_chapter4_2.1.0.42.exe
+		{ CHR,         "english", 447497560,          0, "a6daa6ed56c840f3be7ad6ad920a2f9f2439acc8", ""                                         }, // setup_heroes_chronicles_chapter5_2.1.0.42.exe
+		{ CHR,         "english", 447430456,          0, "93a42dd24453f36e7020afc61bca05b8461a3f04", ""                                         }, // setup_heroes_chronicles_chapter6_2.1.0.42.exe
+		{ CHR,         "english", 481583720,          0, "d74b042015f3c5b667821c5d721ac3d2fdbf43fc", ""                                         }, // setup_heroes_chronicles_chapter7_2.1.0.42.exe
+		{ CHR,         "english", 462976008,          0, "9039050e88b9dabcdb3ffa74b33e6aa86a20b7d9", ""                                         }, // setup_heroes_chronicles_chapter8_2.1.0.42.exe
+	};
+
+	std::string exeHash;
+	std::string binHash;
+	int exeSize = 0;
+	int binSize = 0;
+
+	auto exe = QFile(exeFile);
+	if(exe.open(QFile::ReadOnly)) {
+		QCryptographicHash hash(QCryptographicHash::Algorithm::Sha1);
+		if(hash.addData(&exe))
+			exeHash = hash.result().toHex().toLower().toStdString();
+		else
+			return QString{}; // error with hashing
+		exeSize = exe.size();
+	}
+	else
+		return QString{}; // reading problem
+	if(!binFile.isEmpty())
+	{
+		auto bin = QFile(binFile);
+		if(bin.open(QFile::ReadOnly)) {
+			QCryptographicHash hash(QCryptographicHash::Algorithm::Sha1);
+			if(hash.addData(&bin))
+				binHash = hash.result().toHex().toLower().toStdString();
+			else
+				return QString{}; // error with hashing
+			binSize = bin.size();
+		}
+		else
+			return QString{}; // reading problem
+	}
+	
+	QString hashOutput = tr("SHA1 hash of provided files:\n") + tr("Exe") + " (" + QString::number(exeSize) + " " + tr("bytes") + ")" + ":\n" + QString::fromStdString(exeHash);
+	if(!binHash.empty())
+		hashOutput += "\n" + tr("Bin") + " (" + QString::number(binSize) + " " + tr("bytes") + ")" + ":\n" + QString::fromStdString(binHash);
```

Comment:
Please use strings with placeholders (e.g. `%1`) instead of appending to string. It is both more easy to read in code and easier to translate.